components/camel-whatsapp/src/main/java/org/apache/camel/component/whatsapp/service/WhatsAppServiceRestAPIAdapter.java

```diff
@@ -0,0 +1,285 @@
+	}
+
+	static class OutgoingMediaMessageHandler extends WhatsAppServiceRestAPIAdapter.OutgoingMessageHandler<UploadMediaRequest> {
+
+		public OutgoingMediaMessageHandler(AsyncHttpClient asyncHttpClient, int bufferSize, ObjectMapper mapper,
+				String uri, Class<? extends MessageResponse> returnType) {
+			super(asyncHttpClient, bufferSize, mapper, uri, "multipart/form-data", returnType);
+		}
+
+		public OutgoingMediaMessageHandler(AsyncHttpClient asyncHttpClient, int bufferSize, ObjectMapper mapper,
+				String uri) {
+			this(asyncHttpClient, bufferSize, mapper, uri, MessageResponse.class);
+		}
+
+		@Override
+		protected void addBody(RequestBuilder builder, UploadMediaRequest message) {
+			FilePart filePart = new FilePart("file", message.getUploadMedia().getFile(), message.getUploadMedia().getContentType());
+			builder.setBodyParts(List.of(
+					filePart,
+					new StringPart("messaging_product", "whatsapp")
+			));
+		}
+
+	}
+
+	static class OutgoingPlainMessageHandler extends WhatsAppServiceRestAPIAdapter.OutgoingMessageHandler<BaseMessage> {
+
+		public OutgoingPlainMessageHandler(AsyncHttpClient asyncHttpClient, int bufferSize, ObjectMapper mapper,
+				String uri, Class<? extends MessageResponse> returnType) {
+			super(asyncHttpClient, bufferSize, mapper, uri, "application/json", returnType);
+		}
+
+		public OutgoingPlainMessageHandler(AsyncHttpClient asyncHttpClient, int bufferSize, ObjectMapper mapper,
+				String uri) {
+			this(asyncHttpClient, bufferSize, mapper, uri, MessageResponse.class);
+		}
+
+		@Override
+		protected void addBody(RequestBuilder builder, BaseMessage message) {
+			try {
+				final String body = mapper.writeValueAsString(message);
+				builder.setBody(body);
+			} catch (JsonProcessingException e) {
+				throw new RuntimeCamelException("Could not serialize " + message, e);
+			}
+		}
+
+	}
+
+	abstract static class OutgoingMessageHandler<T extends BaseMessage> {
+		protected final ObjectMapper mapper;
+		private final AsyncHttpClient asyncHttpClient;
+		private final int bufferSize;
+		private final String contentType;
+		private final String uri;
+		private final Class<? extends MessageResponse> resultClass;
+
+		public OutgoingMessageHandler(AsyncHttpClient asyncHttpClient, int bufferSize, ObjectMapper mapper, String uri,
+				String contentType, Class<? extends MessageResponse> resultClass) {
+			this.resultClass = resultClass;
+			this.asyncHttpClient = asyncHttpClient;
+			this.bufferSize = bufferSize;
+			this.mapper = mapper;
+			this.uri = uri;
+			this.contentType = contentType;
+		}
+
+		public void sendMessage(Exchange exchange, AsyncCallback callback, T message, String authorizationToken) {
+			final RequestBuilder builder = new RequestBuilder("POST")
+					.setUrl(uri);
+			if (contentType != null) {
+				builder.setHeader("Content-Type", contentType);
+			}
+			builder.setHeader("Authorization", "Bearer " + authorizationToken);
+			builder.setHeader("Accept", "application/json");
```

Comment:
Header constants, if they are available in one of the libraries in use, can make the code easier to read/maintain.